runtime/vm/jfr.cpp

```diff
@@ -275,14 +377,23 @@ initializeJFR(J9JavaVM *vm)
 static void
 tearDownJFR(J9JavaVM *vm)
 {
+	PORT_ACCESS_FROM_JAVAVM(vm);
 	J9HookInterface **vmHooks = getVMHookInterface(vm);
 
-	// TODO: free global structures
-
+	/* Unhook all the events */
 	(*vmHooks)->J9HookUnregister(vmHooks, J9HOOK_VM_THREAD_CREATED, jfrThreadCreated, NULL);
 	(*vmHooks)->J9HookUnregister(vmHooks, J9HOOK_VM_THREAD_DESTROY, jfrThreadDestroy, NULL);
```

Comment:
The vmThread has been removed from the list, so technically no but because we hold the threadListMutex, the GC can't clean up the object. I'll hook the start/end events for the JFR events and use create/destroy just for the buffer management.